pkg/kubectl/describe.go

```diff
@@ -1105,6 +1106,64 @@ func describeService(service *api.Service, endpoints *api.Endpoints, events *api
 	})
 }
 
+// EndpointsDescriber generates information about an Endpoint.
+type EndpointsDescriber struct {
+	client.Interface
+}
+
+func (d *EndpointsDescriber) Describe(namespace, name string) (string, error) {
+	c := d.Endpoints(namespace)
+
+	ep, err := c.Get(name)
+	if err != nil {
+		return "", err
+	}
+
+	events, _ := d.Events(namespace).Search(ep)
+
+	return describeEndpoint(ep, events)
+}
+
+func describeEndpoint(ep *api.Endpoints, events *api.EventList) (string, error) {
+	return tabbedString(func(out io.Writer) error {
+		fmt.Fprintf(out, "Name:\t%s\n", ep.Name)
+		fmt.Fprintf(out, "Namespace:\t%s\n", ep.Namespace)
+		fmt.Fprintf(out, "Labels:\t%s\n", labels.FormatLabels(ep.Labels))
+
+		fmt.Fprintf(out, "Subsets:\n")
+		for i := range ep.Subsets {
+			subset := &ep.Subsets[i]
+
+			addresses := []string{}
+			for _, addr := range subset.Addresses {
+				addresses = append(addresses, addr.IP)
+			}
+			fmt.Fprintf(out, "\t\tAddresses:\t%s\n", strings.Join(addresses, ","))
+
+			notReadyAddresses := []string{}
+			for _, addr := range subset.NotReadyAddresses {
+				notReadyAddresses = append(notReadyAddresses, addr.IP)
+			}
+			fmt.Fprintf(out, "\t\tNotReadyAddresses:\t%s\n", strings.Join(notReadyAddresses, ","))
+
+			ports := []string{}
+			for _, port := range subset.Ports {
+				name := port.Name
+				if name == "" {
+					name = "<unnamed>"
+				}
+				ports = append(ports, fmt.Sprintf("%s/%d/%s", name, port.Port, port.Protocol))
```

Comment:
Or, you may format endpoints subsets to something like this:
```
Subsets:
  Name    Address    Port    Protocol
  ────    ───────    ────    ────────
  https   172.1.1.1  6443    TCP 
  http    172.1.1.2  80      TCP
```

Refer to:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/resource_printer.go#L470
https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/types.go#L1722-L1742